src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChain.java

```diff
@@ -21,11 +32,31 @@ public ItineraryListFilterChain(List<ItineraryListFilter> filters, boolean debug
             result = filter.filter(result);
         }
 
+        boolean allDeleted = result.stream().allMatch(Itinerary::isFlaggedForDeletion);
+
+        // Add errors, if there were any itineraries, but they were all filtered away
+        if (!itineraries.isEmpty() && allDeleted) {
+            Predicate<Itinerary> isOnStreetAllTheWay = Itinerary::isOnStreetAllTheWay;
+            Predicate<Itinerary> isWorseThanStreet = it -> it.systemNotices.stream().anyMatch(
+                notice -> notice.tag.equals(RemoveTransitIfStreetOnlyIsBetterFilter.TAG));
+            Predicate<Itinerary> isOutsideSearchWindow = it -> it.systemNotices.stream().anyMatch(
+                notice -> notice.tag.equals(LatestDepartureTimeFilter.TAG));
+            if (result.stream().allMatch(isOnStreetAllTheWay.or(isWorseThanStreet))) {
+                routingErrors.add(new RoutingError(WALKING_BETTER_THAN_TRANSIT, null));
```

Comment:
`WALKING_BETTER_THAN_TRANSIT` is not returned for short trips, where a walking itinerary is returned but all the transit trips are removed for being worse than transit. This is due to `allDeleted` being `false`, since the the walk-only itinerary is kept.

![image](https://user-images.githubusercontent.com/58879/140278780-0dec7687-9135-4026-9e39-ab436fdfa405.png)

Should `WALKING_BETTER_THAN_TRANSIT` be returned as an error in this case? If not, when should it be returned?

Reply:
It is returned when no direct mode is specified, however the implementation is up for debate. The other errors are returned only when no results can be found. For this I’m unsure how it should be handled, as a route is not found, but no transit routes.

Reply:
The purpose of the ERROR codes is to help the client to proceed with some action, that action could be to display a message that the best option is to walk - but should the client show this message when the walk option is listed (UC1), or only when the user did not request a direct search(UC2)? To me the UC2 seems most reasonable; hence keep the existing functionality, but no stron opinion.

Reply:
I would expect all of the transit related error messages to have the same behavior (`OUTSIDE_SERVICE_PERIOD`, `NO_STOPS_IN_RANGE`, `WALKING_BETTER_THAN_TRANSIT`, `NO_TRANSIT_CONNECTION_IN_SEARCH_WINDOW`).

`OUTSIDE_SERVICE_PERIOD` and `NO_STOPS_IN_RANGE` are currently returned even if a direct path is found, while the other two are not.

Reply:
Always including the errors could be a way of indicating the reason for the lack of transit results -- even if a direct path was found. That way it is clear what would need to be done to have a transit result if requested.

Reply:
True, I didn't consider that those two produce errors, even tough a walking result is found.

Reply:
This was changed in https://github.com/opentripplanner/OpenTripPlanner/pull/3628/commits/f36185897cf09701b274c4c3a6a08060b4171750